src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -259,6 +277,23 @@ ConvertToDistributedTable(Oid relationId, char *distributionColumnName,
 	char relationKind = 0;
 	Var *distributionColumn = NULL;
 
+	bool streamingReplication PG_USED_FOR_ASSERTS_ONLY =
+		(replicationModel == REPLICATION_MODEL_STREAMING);
+	bool coordinatorReplication PG_USED_FOR_ASSERTS_ONLY =
+		(replicationModel == REPLICATION_MODEL_COORDINATOR);
+	bool tpcReplication PG_USED_FOR_ASSERTS_ONLY =
+		(replicationModel == REPLICATION_MODEL_2PC);
+
+	bool hashDistributed PG_USED_FOR_ASSERTS_ONLY =
+		(distributionMethod == DISTRIBUTE_BY_HASH);
+	bool noneDistributed PG_USED_FOR_ASSERTS_ONLY =
+		(distributionMethod == DISTRIBUTE_BY_NONE);
+
+	Assert(streamingReplication || coordinatorReplication || tpcReplication);
+	Assert(!(streamingReplication && !hashDistributed));
+	Assert((tpcReplication && noneDistributed) ||
+		   ((!tpcReplication && !noneDistributed)));
```

Comment:
Removing all works for me. Asserts crash the database server if they fail, so we should only use them if there is a reason to crash the database server.